### ISSUE TITLE
Thank you for your article 「Devil’s Advocate」, but Is there something wrong in the paper?

### ISSUE THREAD

**Issue by wmathor** [HwiyeolJo/DevilsAdvocate#1]
<img width="640" alt="image" src="https://user-images.githubusercontent.com/32392878/142346095-796285f1-3d78-4585-9408-d5926eb69bb9.png">
There is no problem here and shows that Softmax should be added to the CE loss

----

<img width="651" alt="image" src="https://user-images.githubusercontent.com/32392878/142346027-eedf95db-aa8f-4152-b0f4-15ff17c6a88b.png">

Is there a missing softmax operation here? I think the formula should be as follows:

![MommyTalk1637206265667](https://user-images.githubusercontent.com/32392878/142346815-30a42bed-c05c-40ab-a249-888fcfeb9e58.jpg)

**Comment by HwiyeolJo** [comment]
Thank you for the issue.
Yes, you are right.
We might missed the softmax since Pytorch's cross-entropy function automatically softmax the values.

**Comment by wmathor** [comment]
> Thank you for the issue.
> Yes, you are right.
> We might missed the softmax since Pytorch's cross-entropy function automatically softmax the values.

Thank you for your reply
